src/ImageSharp/Processing/Processors/Transforms/Resize/ResizeProcessorImplementation.cs

```diff
@@ -0,0 +1,184 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.ParallelUtils;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.Memory;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Transforms
+{
+    /// <summary>
+    /// Provides methods that allow the resizing of images using various algorithms.
+    /// Adapted from <see href="http://www.realtimerendering.com/resources/GraphicsGems/gemsiii/filter_rcg.c"/>
+    /// </summary>
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    internal class ResizeProcessorImplementation<TPixel> : TransformProcessorBase<TPixel>
+        where TPixel : struct, IPixel<TPixel>
+    {
+        // The following fields are not immutable but are optionally created on demand.
+        private ResizeKernelMap horizontalKernelMap;
+        private ResizeKernelMap verticalKernelMap;
+
+        private readonly ResizeProcessor parameterSource;
+
+        public ResizeProcessorImplementation(ResizeProcessor parameterSource)
+        {
+            this.parameterSource = parameterSource;
+        }
+
+        /// <summary>
+        /// Gets the sampler to perform the resize operation.
+        /// </summary>
+        public IResampler Sampler => this.parameterSource.Sampler;
+
+        /// <summary>
+        /// Gets the target width.
+        /// </summary>
+        public int Width => this.parameterSource.Width;
```

Comment:
These properties can all be private fields now unless you're expecting inheritance.